### PR TITLE
Feat(Docat): Add Api to upload project icon

### PR DESCRIPTION
**Pull request by reglim**
I created this new pull request because I had some trouble with sqashing the commits and switched to a new branch :)

I didn't create a UI for it yet, because of the
upcoming migration to vue 3.

fixes: #75

### CODE REVIEW

**Comment by reglim** [comment]
Would it make sense to add this feature to docatl as well?

**Comment by randombenj** [comment]
@reglim Good idea, docatl should support this as well :)

File: docat/docat/app.py [hunk 1]
```diff
@@ -113,6 +113,44 @@ def get_project(project):
...
+    db: TinyDB = Depends(get_db),
+):
+    project_base_path = DOCAT_UPLOAD_FOLDER / project
+    icon_path = project_base_path / "logo.png"
+
+    if not project_base_path.exists():
+        response.status_code = status.HTTP_404_NOT_FOUND
+        return ApiResponse(message=f"Project {project} not found")
+
+    if file.content_type != "image/png":
```
Comment (randombenj) [inline docat/docat/app.py:132]: Can't we just check for any type of image? `file.content_type.startswith("image/")` and then save the icon without extention? just `project_base_path / "logo"` The browser will figure out what to display :)

Comment (reglim) [reply docat/docat/app.py:132]: Good idea, I didn't think of that. :+1:

File: docat/docat/app.py [hunk 2]
```diff
@@ -113,6 +113,44 @@ def get_project(project):
...
+    db: TinyDB = Depends(get_db),
+):
+    project_base_path = DOCAT_UPLOAD_FOLDER / project
+    icon_path = project_base_path / "logo"
+
+    if not project_base_path.exists():
+        response.status_code = status.HTTP_404_NOT_FOUND
+        return ApiResponse(message=f"Project {project} not found")
+
+    if not file.content_type.startswith("image/"):
```
Comment (randombenj) [inline docat/docat/app.py:132]: Me uploading the first gif icon:

cat-lover-awesome

**Review by randombenj — APPROVED** [review]
:taco:

**Pull request merged**